mafia

```diff
@@ -52,12 +87,12 @@ exec_mafia () {
 
         bin/bootstrap ) || exit $?
 
-      mkdir -p $(dirname $MAFIA_PATH)
-      cp "$MAFIA_TEMP/mafia/.cabal-sandbox/bin/mafia" "${MAFIA_PATH}.tmp"
-      mv "${MAFIA_PATH}.tmp" "$MAFIA_PATH"
+      cp "$MAFIA_TEMP/mafia/.cabal-sandbox/bin/mafia" "$MAFIA_PATH_TEMP"
+      chmod +x "$MAFIA_PATH_TEMP"
+      mv "$MAFIA_PATH_TEMP" "$MAFIA_PATH"
     }
 
-    exec $MAFIA_PATH $@
+    exec $MAFIA_PATH "$@"
```

Comment:
eyy! does this fix the arguments not being passed to mafia bug I seem to run into sometimes?